src/main/java/org/asamk/signal/commands/ListGroupsCommand.java

```diff
@@ -72,9 +104,86 @@ public int handleCommand(final Namespace ns, final Manager m) {
         List<GroupInfo> groups = m.getGroups();
         boolean detailed = ns.getBoolean("detailed");
 
-        for (GroupInfo group : groups) {
-            printGroup(m, group, detailed);
+        if (ns.getBoolean("json")) {
+            final ObjectMapper jsonProcessor = new ObjectMapper();
+            jsonProcessor.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.ANY);
+            jsonProcessor.disable(JsonGenerator.Feature.AUTO_CLOSE_TARGET);
+
+            if (detailed) {
```

Comment:
hm, I'm not sure if the detailed option is really useful for json output or if the json output should just always output all data.

Reply:
Yeah, non-detailed doesn't really make much sense. I'll force detailed then